DAOS-2721 doc: Update links to command documentation

mjmac opened the pull request:
After the control plane binaries were moved under control/cmd,
a number of links in the documentation needed to be updated.

Signed-off-by: Michael MacDonald <mjmac.macdonald@intel.com>

mjmac commented:
> 
> 
> The link updates you made seem fine and they work. Since you are updating links, can you take a look at the links for 'References to Quickstart guide' which fails to load the src/doc/quickstart.md, and 'daos_shell is a management tool which exercises the client api' which fails to load src/control/dmg/README.md. If you don't want to fix those two links in this PR, that's fine, if you want to make a new PR for it.

I found the second link and fixed that, but I don't see the first one. Can you tell me which file it's in?

jbrosenz commented:
> > The link updates you made seem fine and they work. Since you are updating links, can you take a look at the links for 'References to Quickstart guide' which fails to load the src/doc/quickstart.md, and 'daos_shell is a management tool which exercises the client api' which fails to load src/control/dmg/README.md. If you don't want to fix those two links in this PR, that's fine, if you want to make a new PR for it.
> 
> I found the second link and fixed that, but I don't see the first one. Can you tell me which file it's in?

I wish I could, but I can't even find that line of text when I grep for it.  I don't know where it came from now.  Cue the Twilight Zone music ...

When I looked at the files again, I now encounter these two failed links:

doc/admin/deployment.md
Section: Server Startup 
>>For further details on building and running DAOS see the Quickstart guide.

src/control/server/README.md
Section: Running
>> For instructions on building and running DAOS see the Quickstart guide.

mjmac commented:
> 
> 
> > > The link updates you made seem fine and they work. Since you are updating links, can you take a look at the links for 'References to Quickstart guide' which fails to load the src/doc/quickstart.md, and 'daos_shell is a management tool which exercises the client api' which fails to load src/control/dmg/README.md. If you don't want to fix those two links in this PR, that's fine, if you want to make a new PR for it.
> > 
> > 
> > I found the second link and fixed that, but I don't see the first one. Can you tell me which file it's in?
> 
> I wish I could, but I can't even find that line of text when I grep for it. I don't know where it came from now. Cue the Twilight Zone music ...
> 
> When I looked at the files again, I now encounter these two failed links:
> 
> doc/admin/deployment.md
> Section: Server Startup
> 
> > > For further details on building and running DAOS see the Quickstart guide.
> 
> src/control/server/README.md
> Section: Running
> 
> > > For instructions on building and running DAOS see the Quickstart guide.

I ran the following tool: https://github.com/raviqqe/liche

Found a few more broken links that I'd missed, and fixed those. There are other broken links that it found, but I'll create a separate ticket for those.
